doc/source/data/dataset-internals.rst

```diff
@@ -66,6 +66,8 @@ Execution
 
 This section covers the Datasets execution model and performance considerations.
 
+.. _datasets_lazy_execution:
```

Comment:
I think we should add/edit this page instead of adding a new page.

Reply:
Done.